### ISSUE TITLE
Host header is being set incorrectly, by appending port unnecessarily

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#222]
@bahmutov Related to #215.

This is actually a bug in the underlying `request` library we use to issue HTTP requests. I've opened an issue here: https://github.com/request/request/issues/2372

`request` is appending the `port` to the `Host` header which causes `zeit.co` to return a 404 (like all reverse proxies likely will). That's because `port` should only be appended to the `Host` header when its not `80` or `443`.

You can simulate exactly what `request` is doing under the hood via `curl`.

```bash
## this will return 404
curl -I https://todomvc-express.bahmutov.com:443/app.css --header 'Host: todomvc-express.bahmutov.com:443'
```

```bash
## this will return 200
curl -I https://todomvc-express.bahmutov.com:443/app.css --header 'Host: todomvc-express.bahmutov.com'
```

After splunking through `request's` code I see that if we were to parse out the url ourselves instead of providing as a string, it would work correctly. So we'll do that until they issue a fix.

**Comment by bahmutov** [comment]
Ohh that's pretty good debugging!

**Comment by brian-mann** [comment]
Fixed in [`0.17.3`](https://github.com/cypress-io/cypress/wiki/changelog#0173-09112016).
